lib/nodes/LatchedSwitch.ts

```diff
@@ -0,0 +1,79 @@
+import {BlueshellState, ResultCode, isResultCode, rc, BaseNode, Conditional} from '../models';
+import {Parent} from './Parent';
+import {Constant} from './Constant';
+import {SwitchEntry} from './Switch';
+
+interface AdaptedSwitchEntry<S extends BlueshellState, E> {
+	conditional: Conditional<S, E>,
+	child: BaseNode<S, E>,
+}
+
+/**
+ * Executes the child associated with the first matching conditional.
+ * Returns defaultResult if no conditionals match.
+ *
+ * 11/9/21
+ * @author Timothy Deignan
+ */
+export class LatchedSwitch<S extends BlueshellState, E> extends Parent<S, E> {
```

Comment:
I think this can extend Switch instead of Parent.  You still override onEvent and _afterEvent the way you have, but you wouldn't need your own constructor nor overrides of getChildren or symbol, nor your own private children or entries. You'll need to make entries in Switch protected so you can access it in onEvent here.